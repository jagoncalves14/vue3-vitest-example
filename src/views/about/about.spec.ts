import About from './about.vue'
import { mount } from '@vue/test-utils'

describe('About', () => {
  describe('Snapshots', () => {
    beforeAll(() => {
      vi.useFakeTimers()
      vi.setSystemTime('2022-05-25')
    })

    afterAll(() => {
      vi.useRealTimers()
    })

    it('should mount correctly', () => {
      const wrapper = mount(About)

      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeAll(() => {
      vi.useFakeTimers()
      vi.setSystemTime('2022-05-25')
    })

    afterAll(() => {
      vi.useRealTimers()
    })

    it('eventDate', () => {
      const context = {
        activeMovieId: 0,
      }

      const response = About.methods?.eventDate.call(context)

      expect(response).toBe('5/25/2022')
    })
  })

  describe('Computed', () => {
    it('location', () => {
      const response = About.computed?.location.call({})
      
      expect(response).toBe('Lisboa')
    })
  })
})
