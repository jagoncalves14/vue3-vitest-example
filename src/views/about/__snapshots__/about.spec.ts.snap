// Vitest Snapshot v1

exports[`About > Snapshots > should mount correctly 1`] = `
<div class="about-container">
  <h1>ABOUT</h1>
  <h2>Hi Vue.Lisboa ðŸ‘‹</h2>
  <p>This is a simple <strong>Vite + Vue3 + Vitest App</strong> demo made for 5/25/2022.</p>
</div>
`;
